samples/server/go-server/api/server/storage/project_memstore.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2017 The Grafeas Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"fmt"
+
+	"github.com/grafeas/grafeas/samples/server/go-server/api/server/name"
+	"github.com/grafeas/grafeas/server-go"
+	pb "github.com/grafeas/grafeas/v1alpha1/proto"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// projectMemStore is an in-memory storage solution for Grafeas
+type projectMemStore struct {
```

Comment:
Protect the `.projects` field with a mutex. Otherwise you have to protect it from the caller's side which is much harder to manage (esp between multiple callers).

Reply:
Agree! Fixed in 9f66111. Not sure my solution is best practice so please take an extra look at this.